utils/tsh.c

```diff
@@ -193,6 +201,61 @@ pid_t tsh_run_io(char *cmd, int wait, int *status, int stdin_p[],
 	return pid;
 }
 
+#ifndef DISABLE_LOGSERVER
+static int logserver_subscribe_pipe(int *cmd_pipe, const char *name, int level)
+{
+	errno = 0;
+	if (pipe(cmd_pipe) == -1) {
+		pv_log(ERROR, "cannot create pipe for %s, err: %s", name,
+		       strerror(errno));
+		return -1;
+	}
+
+	pv_logserver_subscribe_fd(cmd_pipe[0], "pantavisor", name, level);
+
+	return 0;
+}
+
+int tsh_run_logserver(char *cmd, int *wstatus, const char *log_source_out, const char *log_source_err)
+{
+	int ret = 0;
+	int out_pipe[2] = { 0 };
+	int err_pipe[2] = { 0 };
+
+	if (logserver_subscribe_pipe(out_pipe, log_source_out, INFO) != 0 ||
+	    logserver_subscribe_pipe(err_pipe, log_source_err, WARN) != 0) {
+		return -1;
+	}
+
+	ret = tsh_run_io(cmd, 1, wstatus, NULL, out_pipe, err_pipe);
```

Comment:
Here I saw something wrong, the third parameter of `tsh_run_io` (wstatus) should be a pointer to that variable